python/GafferUI/Window.py

```diff
@@ -261,14 +261,71 @@ def resizeToFitChild( self, shrink=True, expand=True ) :
 
 		self._qtWidget().resize( s )
 
-	def setPosition( self, position ) :
+	## Repositions the window, attempting to keep the whole frame on screen.
+	# If the window is larger than the available screen area, its position will
+	# be clamped to the top left. If forcePosition is True, the window will be
+	# moved to the specified position even if clipping will occur.
+	def setPosition( self, position, forcePosition = False ) :
 
-		self._qtWidget().move( position.x, position.y )
+		p = QtCore.QPoint( position.x, position.y )
+		if not forcePosition :
+			p = self.__constrainToScreen( p )
+
+		self._qtWidget().move( p )
 
 	def getPosition( self ) :
 
 		return imath.V2i( self._qtWidget().x(), self._qtWidget().y() )
 
+	def __constrainToScreen( self, position ) :
+
+		# Constrain position such that the whole window remains on screen where
+		# possible. This _must_ exist in Qt somewhere, but to date, it has eluded us.
+
+		window = self._qtWidget().windowHandle()
+		if window is None :
+			return position
+
+		screen = window.screen()
+		if screen is None :
+			return position
+
+		# Qt documents that for a window, move() is really pos and includes frame geometry
+
+		screenRect = window.screen().availableGeometry()
+		windowRect = window.frameGeometry()
+
+		# Determine the offset and which direction to move to stay on screen,
+		# based on this size difference and which co-ordinate changed in the
+		# intersected frame
+
+		windowRect.moveTo( position )
+
+		intersection = windowRect.intersected( screenRect )
+		difference = windowRect.size() - intersection.size()
+
+		signX = -1 if intersection.left() == windowRect.left() else 1
+		signY = -1 if intersection.top() == windowRect.top() else 1
+
+		offset = QtCore.QPoint(
+			signX * difference.width(),
+			signY * difference.height()
+		)
```

Comment:
Sanity check : when the window is bigger than the screen, `offset` may contain the sum of the overspill on both left and right edges. But we don't care that it's not longer an actual offset, because the clamping at L323 will kick in anyway, whatever we do?

Reply:
Thats the intention